packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/string_database/index_db/in_memory_string_db.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:collection';
+import 'dart:math';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database_element.dart';
+
+/// {@template amplify_analytics_pinpoint_dart.in_memory_string_db}
+/// Stores string elements in device memory
+/// {@endtemplate}
+class InMemoryStringDb implements StringDatabase {
+  /// {@macro amplify_analytics_pinpoint_dart.in_memory_string_db}
+  InMemoryStringDb({
+    List<StringDatabaseElement> initValues = const [],
+    int lastId = 0,
+  })  : _lastId = lastId,
+        _database = initValues;
+
+  int _lastId = 0;
+  final List<StringDatabaseElement> _database;
```

Comment:
I mentioned in another comment that this store is essentially a queue, with some custom delete behavior. You could consider using `Queue` from dart:collection to hold the data for the in mem impl